WallPaperDev/WallPaperDev/Views/Controllers/SearchImageViewController.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  SearchImageViewController.swift
+//  WallPaperDev
+//
+//  Created by Jamar Gibbs on 11/5/19.
+//  Copyright © 2019 Stephen Ouyang. All rights reserved.
+//
+
+import UIKit
+
+protocol SelectedImageDelegate {
+  func passImageSelected(image: UIImage)
+}
+
+class SearchImageViewController: UIViewController {
+  // MARK: - Custom UIs
+  private lazy var createImageButton = BigBlueButton(frame: .zero)
+  private lazy var searchImagesCV = SearchImagesCV(frame: .zero, collectionViewLayout: SearchImagesCVLayout())
+  private lazy var emptyView = SelectedGoalsEmptyView()
+  private lazy var searchController = UISearchController(searchResultsController: nil)
+  private let goalsVC = GoalsSelectionViewController()
+  
+  private let viewModel = SearchImagesViewModel()
+  private let selectImageViewModel = SelectionViewModel()
+  weak var coordinator: MainCoordinator?
+  var selectedImageDelegate : SelectedImageDelegate?
+  
+  override var preferredStatusBarStyle: UIStatusBarStyle {
+    .lightContent
+  }
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    self.view.backgroundColor = .white
+    setupViews()
+  }
+  
+  private func setupViews() {
+    setupNavBar()
+    setupImageCollectionView()
+    setupBlueButton()
+  }
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    navigationController?.navigationBar.isHidden = false
+  }
+  
+  private func setupImageCollectionView() {
+    self.view.addSubview(searchImagesCV)
+    searchImagesCV.delegate = self
+    searchImagesCV.dataSource = self
+    NSLayoutConstraint.activate([
+      searchImagesCV.leadingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.leadingAnchor),
+      searchImagesCV.trailingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.trailingAnchor),
+      searchImagesCV.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
+      searchImagesCV.bottomAnchor.constraint(equalTo: self.view.bottomAnchor)
+    ])
+  }
+  
+  private func setupNavBar() {
+    searchController.searchBar.delegate = self
+    searchController.searchBar.placeholder = "Search for images"
+    searchController.searchBar.autocorrectionType = .default
+    searchController.searchBar.sizeToFit()
+    
+    navigationItem.title = "Search for a new image"
```

Comment:
Let's toss these into the localizable file

Reply:
I think it will look better if I remove the title since it reduce the number of elements in the nav bar. Thus avoid the cramped look.